polaris/ocean/convergence/spherical/analysis.py

```diff
@@ -0,0 +1,332 @@
+    """
+    A step for analyzing the output from the geostrophic convergence test
+
+    Attributes
+    ----------
+    resolutions : list of float
+        The resolutions of the meshes that have been run
+
+    icosahedral : bool
+        Whether to use icosahedral, as opposed to less regular, JIGSAW
+        meshes
+
+    dependencies_dict : dict of dict of polaris.Steps
+        The dependencies of this step
+    """
+    def __init__(self, component, resolutions, icosahedral, subdir,
+                 dependencies, convergence_vars):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component the step belongs to
+
+        resolutions : list of float
+            The resolutions of the meshes that have been run
+
+        icosahedral : bool
+            Whether to use icosahedral, as opposed to less regular, JIGSAW
+            meshes
+
+        subdir : str
+            The subdirectory that the step resides in
+
+        dependencies : dict of dict of polaris.Steps
+            The dependencies of this step
+        """
+        super().__init__(component=component, name='analysis', subdir=subdir)
+        self.resolutions = resolutions
+        self.icosahedral = icosahedral
+        self.dependencies_dict = dependencies
+        self.convergence_vars = convergence_vars
+
+        for _, var in convergence_vars.items():
+            self.add_output_file(f'convergence_{var["name"]}.png')
+
+    def setup(self):
+        """
+        Add input files based on resolutions, which may have been changed by
+        user config options
+        """
+        dependencies = self.dependencies_dict
+
+        for resolution in self.resolutions:
+            mesh_name = resolution_to_subdir(resolution)
+            base_mesh = dependencies['mesh'][resolution]
+            init = dependencies['init'][resolution]
+            forward = dependencies['forward'][resolution]
+            self.add_input_file(
+                filename=f'{mesh_name}_mesh.nc',
+                work_dir_target=f'{base_mesh.path}/base_mesh.nc')
+            self.add_input_file(
+                filename=f'{mesh_name}_init.nc',
+                work_dir_target=f'{init.path}/initial_state.nc')
+            self.add_input_file(
+                filename=f'{mesh_name}_output.nc',
+                work_dir_target=f'{forward.path}/output.nc')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        plt.switch_backend('Agg')
+        convergence_vars = self.convergence_vars
+        for _, var in convergence_vars.items():
+            self.plot_convergence(var)
+
+    def plot_convergence(self, convergence_field):
+        """
+        Create a convergence plot
+
+        Parameters
+        ----------
+        convergence_field: dict
+            Dict containing attributes of the field of which to evaluate
+            convergence
```

Comment:
This needs an explanation of what the expected keys and values are of `convergence_field`.  It might be simpler to take these parameters separately here and unpack the dictionary above in `run()` instead.  That would be my preference.